twinkle.js

```diff
@@ -262,122 +262,10 @@ Twinkle.addPortlet = function(navigation, id, text, type, nextnodeid) {
-			}
-
-			outerNavClass = 'mw-portlet vector-menu';
-			if (navigation === 'mw-panel') {
-				outerNavClass += ' vector-menu-portal';
-			} else if (type === 'menu') {
-				outerNavClass += ' vector-menu-dropdown vector-dropdown vector-menu-dropdown-noicon';
-			} else {
-				outerNavClass += ' vector-menu-tabs';
-			}
-
-			innerDivClass = 'vector-menu-content vector-dropdown-content';
-			break;
-		case 'modern':
-			if (navigation !== 'mw_portlets' && navigation !== 'mw_contentwrapper') {
-				navigation = 'mw_portlets';
-			}
-			outerNavClass = 'portlet';
-			break;
-		case 'timeless':
-			outerNavClass = 'mw-portlet';
-			innerDivClass = 'mw-portlet-body';
-			break;
-		default:
-			navigation = 'column-one';
-			outerNavClass = 'portlet';
-			break;
-	}
-
-	// Build the DOM elements.
-	var outerNav, heading;
-	if (skin === 'vector-2022') {
-		outerNav = document.createElement('div');
-		heading = document.createElement('label');
-	} else {
-		outerNav = document.createElement('nav');
-		heading = document.createElement('h3');
+	if (type === 'menu') {
+		nextnodeid = 'p-cactions';
 	}
-
-	outerNav.setAttribute('aria-labelledby', id + '-label');
-	outerNav.className = outerNavClass + ' emptyPortlet';
-	outerNav.id = id;
-	if (nextnode && nextnode.parentNode === root) {
-		root.insertBefore(outerNav, nextnode);
-	} else {
-		root.appendChild(outerNav);
-	}
-
-	heading.id = id + '-label';
-	var ul = document.createElement('ul');
-
-	if (skin === 'vector' || skin === 'vector-2022') {
-		heading.setAttribute('for', id + '-dropdown-checkbox');
-		ul.className = 'vector-menu-content-list';
-		heading.className = 'vector-menu-heading vector-dropdown-label';
-
-		// add invisible checkbox to keep menu open when clicked
-		// similar to the p-cactions ("More") menu
-		if (outerNavClass.indexOf('vector-menu-dropdown') !== -1) {
-			var chkbox = document.createElement('input');
-			chkbox.id = id + '-dropdown-checkbox';
-			chkbox.className = 'vector-menu-checkbox vector-dropdown-checkbox';
-			chkbox.setAttribute('type', 'checkbox');
-			chkbox.setAttribute('aria-labelledby', id + '-label');
-			outerNav.appendChild(chkbox);
-
-			// Vector gets its title in a span; all others except
-			// timeless have no title, and it has no span
-			var span = document.createElement('span');
-			span.appendChild(document.createTextNode(text));
-			heading.appendChild(span);
-
-			var a = document.createElement('a');
-			a.href = '#';
-
-			$(a).click(function(e) {
-				e.preventDefault();
-			});
-
-			heading.appendChild(a);
-		}
-	} else {
-		// Basically just Timeless
-		heading.appendChild(document.createTextNode(text));
-	}
-
-	outerNav.appendChild(heading);
-
-	if (innerDivClass) {
-		var innerDiv = document.createElement('div');
-		innerDiv.className = innerDivClass;
-		innerDiv.appendChild(ul);
-		outerNav.appendChild(innerDiv);
-	} else {
-		outerNav.appendChild(ul);
-	}
-
-	return outerNav;
+	return mw.util.addPortlet(id, text, '#' + nextnodeid);
```

Comment:
With this patch, the Twinkle menu has moved from the right side of the Tools/More menu, to the left side. Is that an easy fix to make it show up on the right side?

Old:
<img width="415" alt="image" src="https://github.com/wikimedia-gadgets/twinkle/assets/79697282/67c47cc7-3592-4cda-9900-7dc8d8e89416">

New:
<img width="429" alt="image" src="https://github.com/wikimedia-gadgets/twinkle/assets/79697282/12e004a4-0975-4edd-b1b1-929ddb9d00ed">

Reply:
This is one of the constraints the Vector API has - for dropdowns you can only inject in front of the #p-cactions element - it's how we signal you want a dropdown.  In Vector 2022 it was also intentional given the pinnable functionality - the dropdown for more menu should be the furthest right since it can be moved to the right hand side. 

You could add additional code to move the portlet after it's inserted if you are okay with this complexity like so:
```
const portlet = mw.util.addPortlet(id, text, '#' + nextnodeid);
$(portlet).insertAfter('#p-cactions')
```

Reply:
I just tested it out. `$(portlet).insertAfter('#p-cactions')` only works in Vector 2010. Would need separate code for V22 (`$(portlet).insertAfter('.vector-page-tools-landmark')` maybe), which combined with the conditional logic introduces a lot of complexity to this otherwise simple patch. Let's just keep it on the left for now and hopefully users are OK with it.